frontend/app/routes/$lang+/_public+/apply+/$id+/review-information.tsx

```diff
@@ -40,6 +40,14 @@ export async function loader({ request, params }: LoaderFunctionArgs) {
   const parsedDateOfBirthString = parseDateString(state.dateOfBirth ?? '');
   const dateOfBirth = new Date(Number.parseInt(parsedDateOfBirthString.year ?? ''), Number.parseInt(parsedDateOfBirthString.month ?? ''), Number.parseInt(parsedDateOfBirthString.day ?? ''));
   const partnerDob = { year: state.partnerInformation?.year ?? 2024, month: state.partnerInformation?.month ?? 1, day: state.partnerInformation?.day ?? 1 };
+  // Getting province by Id
+  const allRegions = await getLookupService().getAllRegions();
+  const provinceMailing = allRegions.filter((region) => region.provinceTerritoryStateId === state.personalInformation?.mailingProvince)[0];
+  const provinceHome = allRegions.filter((region) => region.provinceTerritoryStateId === state.personalInformation?.homeProvince)[0];
+  // Getting Country by Id
+  const allCountries = await getLookupService().getAllCountries();
+  const countryMailing = allCountries.filter((country) => country.countryId === state.personalInformation?.mailingCountry)[0];
+  const countryHome = allCountries.filter((country) => country.countryId === state.personalInformation?.homeCountry)[0];
```

Comment:
```suggestion
  const countryMailing = allCountries.find((country) => country.countryId === state.personalInformation?.mailingCountry);
  const countryHome = allCountries.find((country) => country.countryId === state.personalInformation?.homeCountry);
```

We should also log and throw an `Error` if `countryMailing` or `countryHome` cannot be found since we expect a home/mailing country to exist. 

We also wouldn't have to deal with the undefined case when rendering the component if we throw early here.